db.go

```diff
@@ -128,6 +128,106 @@ func SmartUnmarshal[I any](respond interface{}, wrapperError error) (data I, err
 	return data, err
 }
 
+// SmartUnmarshalAll takes an interface input and unmarshals it into the given
+// type, traversing the nested slices and maps, regardless of the input being
+// wrapped in RawQuery or not. When any error is seen during the unmarshaling,
+// it would keep the return value along with the error message.
+func SmartUnmarshalAll[I any](input interface{}) ([]I, error) {
+	// NOTE: We could use a dedicated error type to provide unmarshal failure,
+	// which may allow users to ignore some erroneous data being returned from
+	// the database.
+	return handleInterfaces[I](input)
+}
+
+func handleInterfaces[I any](input interface{}) ([]I, error) {
+	var result []I
+	var errs []error
+	inputs, ok := input.([]interface{})
+	if !ok {
+		return handleInput[I](input)
+	}
+
+	for _, i := range inputs {
+		data, err := handleInterfaces[I](i)
+		errs = append(errs, err)
+		result = append(result, data...)
+	}
+	if len(errs) > 0 {
+		return result, errors.Join(errs...)
```

Comment:
It's a good point, as a library to be used by various use cases, I mimicked the official implementation (with slightest tweak so that the code flow is still easier to follow) with d927cf0
https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/errors/join.go

Reply:
Now the code chunks a bit messier than I like, it may be worth porting out some of those utility structs and vars out to a separate file?

Reply:
Hey, as SurrealDB.go moved to 1.20 we could go back.
I'm sorry this I didn't think we would move that quickly.